### PR TITLE
Use System32\tar.exe if present instead of 7zip for zip archives.

### PR DESCRIPTION
**Pull request by BillyONeal**
This is a more targeted hotfix version of https://github.com/microsoft/vcpkg-tool/pull/404 as there are outstanding code review discussions there. (I do still intend to apply the changes therein but don't want code reviewers to feel like they can't say anything because it's resolving a high priority issue)

### CODE REVIEW

**Comment by Neumann-A** [comment]
This PR + #405 + #407. Should solve all msiexec issues and completely move from 7z to tar. 

- #407: makes 7z install without msiexec possible if 7z is required due to not having `tar`. This only leaves python2/kinectsdk2/msmpi to require msiexec. Probably also update the search path for `7z` in `vcpkg_find_acquire_program`. Only `libpng` seems to require it (maybe change to `cmake -E tar`?). 
- #405: makes it possible to extract the 7z archive from an 7z sfx installer. This is required to make PortableGit extract without the need for 7z.
- This PR enables the usage of tar to unzip cmake which then can be used to decompress 7z archives via `cmake -E tar` eliminating the need for 7z completely.

File: src/vcpkg/base/system.cpp [hunk 1]
```diff
@@ -273,6 +273,28 @@ namespace vcpkg
...
+            {
+                return std::string("Expected the SystemRoot environment variable to be always set on Windows.");
+            }
+        }();
+        return s_system_root;
+    }
+
+    const ExpectedS<Path>& get_system32() noexcept
+    {
+        static const ExpectedS<Path> s_system32 = get_system_root().map([](const Path& p) { return p / "system32"; });
```
Comment (strega-nil) [inline src/vcpkg/base/system.cpp:295]:
```suggestion
        static const ExpectedS<Path> s_system32 = get_system_root().map([](const Path& p) { return p / "System32"; });
```
I'd prefer it to use the canonical casing (even if it doesn't really matter)

File: include/vcpkg/tools.h [hunk 2]
```diff
@@ -30,15 +31,9 @@ namespace vcpkg
...
-    enum class RequireExactVersions
-    {
-        YES,
-        NO,
-    };
-
     struct ToolCache
     {
-        virtual ~ToolCache() { }
+        virtual ~ToolCache();
```
Comment (strega-nil) [inline include/vcpkg/tools.h:36]:
```suggestion
        virtual ~ToolCache() = default;
```
no reason to put this in the C++ file I think

File: src/vcpkg/archives.cpp [hunk 3]
```diff
@@ -183,6 +197,15 @@ namespace vcpkg
         Checks::check_exit(VCPKG_LINE_INFO, code == 0, "tar failed while extracting %s", archive);
     }
 
+    void extract_tar_cmake(const Path& cmake_tool, const Path& archive, const Path& to_path)
+    {
+        // Note that CMake's built in tar can extract more archive types than many system tars; e.g. 7z
+        const auto code =
+            cmd_execute(Command{cmake_tool}.string_arg("-E").string_arg("tar").string_arg("xzf").path_arg(archive),
+                        WorkingDirectory{to_path});
+        Checks::check_exit(VCPKG_LINE_INFO, code == 0, "tar failed while extracting %s", archive);
```
Comment (strega-nil) [inline src/vcpkg/archives.cpp:206]: Should this use localized message?

**Review by strega-nil — APPROVED** [review]
I definitely appreciate fewer changes

File: src/vcpkg/archives.cpp (hunk 3 above)
Comment (BillyONeal) [reply src/vcpkg/archives.cpp:206]: I made everything localized in the other PR but this PR is intentionally the minimal change.

**Review by ras0219-msft — APPROVED** [review]
Thanks a bunch for extracting the hotfix!

**Pull request merged**